### ISSUE TITLE
`consistent-type-specifier-style`: rename `prefer-top-level`

### ISSUE THREAD

**Issue by privatenumber** [un-ts/eslint-plugin-import-x#221]
## Problem

Currently, the [`consistent-type-specifier-style`](https://github.com/un-ts/eslint-plugin-import-x/blob/master/docs/rules/consistent-type-specifier-style.md) rule has options `prefer-top-level` and `prefer-inline`.

While it uses the word "prefer", it doesn't seem to be preferential—but more of a requirement—as `prefer-top-level` errors on:
```ts
import { valueA, type typeB } from './file'
```

## Proposal

1. Rename the current options to `top-level` and `inline`
2. Add a new `prefer-top-level` that only errors if it's possible to move the `type` to the top-level without causing a duplicate import:
	
```ts
	// Passes
	import { valueA, type typeB } from './file' // can be inline because making it top-level will require a separate import for valueA
	
	// Errors
	import { type typeB } from './file' // type should be top-level
	```

**Comment by SukkaW** [comment]
We can add `top-level` and `inline` as aliases for now. Changing the `prefer-top-level` would be a breaking change.
